core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -827,10 +832,57 @@ private[spark] class BlockManager(
    */
   private[spark] def getRemoteValues[T: ClassTag](blockId: BlockId): Option[BlockResult] = {
     val ct = implicitly[ClassTag[T]]
-    getRemoteManagedBuffer(blockId).map { data =>
+    getRemoteBlock(blockId, (data: ManagedBuffer) => {
       val values =
         serializerManager.dataDeserializeStream(blockId, data.createInputStream())(ct)
       new BlockResult(values, DataReadMethod.Network, data.size)
+    })
+  }
+
+  /**
+   * Get the remote block and transform it to the provided data type.
+   *
+   * If the block is persisted to the disk and stored at an executor running on the same host then
+   * first it is tried to be accessed using the local directories of the other executor directly.
+   * If the file is successfully identified then tried to be transformed by the provided
+   * transformation function which expected to open the file. If there is any exception during this
+   * transformation then block access falls back to fetching it from the remote executor via the
+   * network.
+   *
+   * @param blockId identifies the block to get
+   * @param bufferTransformer this transformer expected to open the file if the block is backed by a
+   *                          file by this it is guaranteed the whole content can be loaded
+   * @tparam T result type
+   * @return
+   */
+   private[spark] def getRemoteBlock[T](
+       blockId: BlockId,
+       bufferTransformer: ManagedBuffer => T): Option[T] = {
```

Comment:
nit: its a bit more idiomatic in scala to make this function in a separate parameter list, which makes the places you use it look a bit more natural:

```scala
private[spark] def getRemoteBlock[T](
    blockId: BlockId)
    (bufferTransformer: ManagedBuffer => T): Option[T] = {
...

// then used like this:
getRemoteBlock(blockId) { data => }
```

Reply:
Of course I know the multiple parameter lists. I just followed here this rule from the style guide: [Do NOT use multi-parameter lists.](https://github.com/databricks/scala-style-guide#java-multi-param-list)

Reply:
oh I guess I didn't know that style rule myself, thanks for correction

Reply:
I have seen cases for violating this but they seamed to me more generic methods (Loner patterns).